### PR TITLE
Create mapping interface for span events & attributes

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Creates an interface for converting POJOs to span events/attributes. I've created a simple Kotlin class that represents the values required to create a span/event.

## Testing

Updated existing unit test coverage.

### CODE REVIEW

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/SpanAttributeMapper.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package io.embrace.android.embracesdk.arch
+
+/**
+ * Converts an object of type T to a [SpanAttributeData]
+ */
+internal fun interface SpanAttributeMapper<T> {
```
Comment (lucaslabari) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/SpanAttributeMapper.kt:6]: These two mappers are not being used. Are they gonna be used in another PR?

**Review by lucaslabari — APPROVED** [review]
LGTM. Just left a question about the mappers that are not yet being used.

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/SpanAttributeMapper.kt (hunk 1 above)
Comment (fractalwrench) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/SpanAttributeMapper.kt:6]: Yep - I've got an example of how they work in #406.

**Review by bidetofevil — APPROVED** [review]
Really nice!
